src/main/java/core/basesyntax/service/impl/DataWriterServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.DataWriterService;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class DataWriterServiceImpl implements DataWriterService {
+    private static final String CSV_HEADER = "fruit,quantity";
+    private static final String SEPARATOR = ",";
+
+    @Override
+    public void writeReportToFile(Map<String, Integer> report, String fileName) {
+        try (FileWriter writer = new FileWriter(fileName)) {
+            writer.write(CSV_HEADER + System.lineSeparator());
+
```

Comment:
separate concerns, pass a prepared report String to this method, generate in a separate service